server/settings/publish_plugins.py

```diff
@@ -1244,6 +1256,17 @@ class PublishPuginsModel(BaseSettingsModel):
         default_factory=ExtractThumbnailModel,
         title="Extract Thumbnail"
     )
+    ExtractThumbnailFromSource: ExtractThumbnailFromSourceModel = SettingsField(  # noqa: E501
+        default_factory=ExtractThumbnailFromSourceModel,
+        title="Extract Thumbnail (from source)",
+        description=(
+            "Extract thumbnails from explicit file set in "
+            "instance.data['thumbnailSource'] using ffmpeg "
+            "and oiiotool."
+            "Used when host does not provide thumbnail, but artist could set "
+            "custom thumbnail source file. (TrayPublisher, Webpublisher)"
```

Comment:
It is not only for traypublisher and webpublisher, it can be used everywhere.
```suggestion
            "custom thumbnail source file."
```